.github/workflows/backup-kubernetes-databases-inhouse.yml

```diff
@@ -0,0 +1,168 @@
+        required: false
+        default: '["self-hosted", "Linux", "X64", "bratislava"]'
+      backup:
+        description: 'Two types of backups postgress or shared-wordpress'
+        type: string
+        required: false
+        default: 'postgres'
+      cluster:
+        description: 'Kubernetes cluster name'
+        required: true
+        type: string
+      namespace:
+        description: 'Namespace where should be build app deployed'
+        required: true
+        type: string
+      url:
+        description: 'Kubernetes cluster url'
+        default: 'https://tkg.dev.bratislava.sk'
+        required: true
+        type: string
+    secrets:
+      service-account:
+        description: 'Kubernetes service account'
+        required: true
+      minio-endpoint:
+        description: 'Minio endpoint url'
+        required: true
+      minio-access:
+        description: 'Minio access key'
+        required: true
+      minio-secret:
+        description: 'Minio secret key'
+        required: true
+      minio-bucket:
+        description: 'Minio bucket name'
+        required: true
+      aws-key:
+        description: 'AWS key id'
+        required: true
+      aws-secret:
+        description: 'AWS secret access key'
+        required: true
+      aws-bucket:
+        description: 'AWS bucket name'
+        required: true
+
+jobs:
+  backup:
+    name: Backup databases
+    runs-on: ${{fromJSON(inputs.runs-on)}}
+    steps:
+      - name: Checking out
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Pipelines Version
+        run: |
+          echo "Pipelines version: 2.1.0"
+
+      - name: Directory check
+        run: pwd
+
+      - name: Kubectl tool installer
+        uses: Azure/setup-kubectl@v4.0.0
+
+      - name: Kubernetes set service account token
+        run: kubectl config set-credentials default --token=${{ secrets.service-account }}
+
+      - name: Kubernetes set server with certificate account token
+        run: kubectl config set-cluster ${{ inputs.cluster }}  --insecure-skip-tls-verify --server=${{ inputs.url }}
+
+      - name: Kubernetes set context cluster
+        run: kubectl config set-context ${{ inputs.cluster }} --cluster=${{ inputs.cluster }} --user=default
+
+      - name: Kubernetes use context
+        run: kubectl config use-context ${{ inputs.cluster }}
+      - name: Run database backup
+        run: |    
+          backup="${{ inputs.backup }}"
+          
+          if [[ $backup == "shared-wordpress" ]]; then
+            echo "We are going to backup shared wordpress databases."
+            select=$(kubectl get pod -n ${{ inputs.namespace }} --insecure-skip-tls-verify -l app.kubernetes.io/name=wordpress -o name)
+          else 
+            echo "We are going to backup postgres databases."
+            select=$(kubectl get pod -n ${{ inputs.namespace }} --insecure-skip-tls-verify -l service=database -o name)
+          fi
```

Comment:
As both commands are the same and only difference is the label, I would rather see just the label in the `if` and the actual command pull out from it. It is to keep a bit more DRY and not repeat ourselves. Also, when we would need to change the command we can change it only in one place.

Something like (you can skip oneliner `if`s)
```bash
if [[ "$backup" == "shared-wordpress" ]]; then label="app.kubernetes.io/name=wordpress"; else label="service=database"; fi
select=$(kubectl get pod --namespace "${{ inputs.namespace }}" --insecure-skip-tls-verify --selector "$label" --output name)
```

There are couple more things:
 - Generally speaking in scripts we prefer long version of options, e.g., `--namespace` rather then just `-n`. It is mostly due to [readability of scripts](https://stackoverflow.com/a/16676385)
 - When using bash variable, especially in condition, you *(in most cases)* want to put double quotes around it, e.g., `if [[ $backup == "shared-wordpress" ]]` becomes `if [[ "$backup" == "shared-wordpress" ]]`. It is just a convention due to [this](https://www.shellcheck.net/wiki/SC2086), [this](https://www.shellcheck.net/wiki/SC2248) or [this](https://unix.stackexchange.com/a/86802)